### PR TITLE
highlights(lua): add spell

### PR DESCRIPTION
**Pull request by lewis6991**

### CODE REVIEW

File: queries/lua/highlights.scm [hunk 1]
```diff
@@ -181,12 +181,14 @@
...
 
 (comment) @comment
+(comment) @spell
 
 (hash_bang_line) @comment
 
 (number) @number
 
 (string) @string
+(string) @spell
```
Comment (vigoux) [inline queries/lua/highlights.scm:191]: I never quite liked spellchecking within strings... But if the people wants that, then let's have it.

Comment (lewis6991) [reply queries/lua/highlights.scm:191]: Me neither, but I think we are both in the minority.

We could add an option for it.

Comment (clason) [reply queries/lua/highlights.scm:191]: Options for queries are a general todo. (Part of the "upstream and generalize predicate handling" tracking issue, now defered to 0.9.)

I'd like a general solution for such optional captures (e.g., for the fancier conceals) first.

Until then, we should do this consistently all-or-nothing.

Comment (lewis6991) [reply queries/lua/highlights.scm:191]: Can someone approve this then?

**Review by clason — APPROVED** [review]

File: queries/lua/highlights.scm (hunk 1 above)
Comment (clason) [reply queries/lua/highlights.scm:191]: Sure. It's consistent with upstream highlights, which is the main concern for me.

Comment (ram02z) [reply queries/lua/highlights.scm:191]: Also, not a fan of the spellchecking within strings, how can I override this query locally?

**Pull request merged**